src/com/google/cose/MacKey.java

```diff
@@ -51,7 +50,87 @@ private MacKey(final DataItem cborKey) throws CborException, CoseException {
         && operations.contains(Headers.KEY_OPERATIONS_MAC_VERIFY))) {
       return;
     }
-    throw new CoseException("Encryption key requires encrypt and decrypt operations.");
+    throw new CoseException("Mac key requires create mac and verify mac operations.");
+  }
+
+  static class Builder {
+    private String keyId;
+    private Algorithm algorithm;
+    private final List<Integer> operations;
+    private byte[] baseIv;
+    private byte[] secretKey;
+
+    Builder() {
+      operations = new ArrayList<>();
+    }
+
+    public MacKey build() throws CoseException, CborException {
+      if (secretKey == null) {
+        throw new CoseException("Need key material information.");
+      }
+
+      if (operations.size() != 0 && !operations.contains(Headers.KEY_OPERATIONS_MAC_CREATE)
+          && !operations.contains(Headers.KEY_OPERATIONS_MAC_VERIFY)) {
+        throw new CoseException("Need CreateMac and VerifyMac operation for MAC key.");
+      }
+
+      Map cborKey = new Map();
+      cborKey.put(new UnsignedInteger(Headers.KEY_PARAMETER_KEY_TYPE),
+          new UnsignedInteger(Headers.KEY_TYPE_SYMMETRIC));
+
+      if (keyId != null) {
```

Comment:
Can these repeated portions of the build across the different key classes be moved to the superclass?